internal/app/grpcclient/impl.go

```diff
@@ -164,8 +174,19 @@ func (i *invoker) serverStream(
 		}
 	}()
 
+	timing, err := internal.GetCancelTiming(ccr.Cancel)
+	if err != nil {
+		return nil, err
+	}
+	// If the cancel timing specifies after 0 responses, then cancel before
+	// receiving anything
+	if timing.AfterNumResponses == 0 {
+		ctxCancel()
+	}
+	totalRcvd := 0
 	for {
 		msg, err := stream.Recv()
+		totalRcvd++
```

Comment:
For some reason, if `AfterNumResponses == 0` and the context was cancelled, the `stream.Recv` call above returns a message and a `nil` error on the first iteration.